database/src/procedures/api_patch_system_user.ts

```diff
@@ -0,0 +1,109 @@
+import { Knex } from 'knex';
+
+/**
+ * Add/update function that patches a system user record.
+ *
+ * Steps:
+ * 1. Attempts to find an existing system user record
+ *    - First using `p_system_user_guid`
+ *    - Second using `p_user_identifier` and `p_user_identity_source_name`
+ * 2. If no user is found, return null
+ * 3. If a user is found, update the system user record with the latest information passed to this function if any of
+ * the incoming values are not the same as the existing values
+ *
+ * @export
+ * @param {Knex} knex
+ * @return {*}  {Promise<void>}
+ */
+export async function up(knex: Knex): Promise<void> {
+  await knex.raw(`--sql
+    set search_path=biohub;
+
+    CREATE OR REPLACE FUNCTION
+      api_patch_system_user (
+        p_system_user_guid character varying,
+        p_user_identifier character varying,
+        p_user_identity_source_name character varying,
+        p_email character varying,
+        p_display_name character varying,
+        p_given_name character varying,
+        p_family_name character varying,
+        p_agency character varying
+      )
+      RETURNS integer
+      LANGUAGE plpgsql
+      SET client_min_messages TO 'warning'
+      AS $$
+        -- *******************************************************************
+        -- Procedure: api_patch_system_user
+        -- Purpose: Updates a system_user record if any of the incoming values are not the same as the existing values.
+        --
+        -- MODIFICATION HISTORY
+        -- Person           Date        Comments
+        -- ---------------- ----------- --------------------------------------
+        -- nick.phura@quartech.com
+        --                  2023-08-01  initial release
+        -- *******************************************************************
+        DECLARE
+          _system_user system_user%rowtype;
+          _user_identity_source_id user_identity_source.user_identity_source_id%type;
+        BEGIN
+          -- Attempt to find user based on guid
+          SELECT * INTO _system_user FROM system_user
+            WHERE LOWER(user_guid) = LOWER(p_system_user_guid)
```

Comment:
Added these `LOWER(...)` calls her and below to make sure the string comparisons are case-insensitive.